lib/web/apiserver_test.go

```diff
@@ -6482,3 +6488,156 @@ func init() {
 		&metav1.Status{},
 	)
 }
+
+// TestForwardingTraces checks that the userContext includes the ID of the
+// access request after it has been consumed and the web session has been renewed.
+func TestForwardingTraces(t *testing.T) {
+	t.Parallel()
+
+	env := newWebPack(t, 1)
+	p := env.proxies[0]
+
+	span := &tracepb.TracesData{
+		ResourceSpans: []*tracepb.ResourceSpans{
+			{
+				Resource: &resourcev1.Resource{
+					Attributes: []*commonv1.KeyValue{
+						{
+							Key: "test",
+							Value: &commonv1.AnyValue{
+								Value: &commonv1.AnyValue_IntValue{
+									IntValue: 0,
+								},
+							},
+						},
+					},
+				},
+				ScopeSpans: []*tracepb.ScopeSpans{
+					{
+						Spans: []*tracepb.Span{
+							{
+								TraceId:           []byte{1, 2, 3, 4},
+								SpanId:            []byte{5, 6, 7, 8},
+								TraceState:        "",
+								ParentSpanId:      []byte{9, 10, 11, 12},
```

Comment:
Are the IDs here arbitrary? Are they hex-encoded? Please document or generate in code, so we can tell where the values come from.

Reply:
Ack, maybe add a small comment saying that?